purchases/src/main/kotlin/com/revenuecat/purchases/galaxy/conversions/StoreProductConversions.kt

```diff
@@ -3,16 +3,31 @@ package com.revenuecat.purchases.galaxy.conversions
 import com.revenuecat.purchases.ProductType
 import com.revenuecat.purchases.common.LogIntent
 import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.galaxy.GalaxyPurchasingData
 import com.revenuecat.purchases.galaxy.GalaxyStoreProduct
 import com.revenuecat.purchases.galaxy.GalaxyStrings
+import com.revenuecat.purchases.galaxy.GalaxySubscriptionOption
 import com.revenuecat.purchases.models.Period
 import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
 import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.SubscriptionOption
+import com.revenuecat.purchases.models.SubscriptionOptions
 import com.samsung.android.sdk.iap.lib.vo.ProductVo
+import com.samsung.android.sdk.iap.lib.vo.PromotionEligibilityVo
+import java.util.HashSet
 
-internal fun ProductVo.toStoreProduct(): StoreProduct {
-    val type = this.type.createRevenueCatProductTypeFromSamsungIAPTypeString()
-    val period: Period? = if (type == ProductType.SUBS) {
+private const val ELIGIBILITY_PRICING_FREE_TRIAL = "FreeTrial"
+private const val ELIGIBILITY_PRICING_TIERED_PRICE = "TieredPrice"
+
+internal fun ProductVo.toStoreProduct(
+    promotionEligibilities: List<PromotionEligibilityVo>? = null,
```

Comment:
You may be wondering why `ProductVo.storeProduct` now takes in a `List<PromotionEligibilityVo>?` instead of a single `PromotionEligibilityVo?`. This is because the `IapHelper.getPromotionEligibility()`'s listener provides us with an ArrayList of PromotionEligibilityVo? objects:

```java
public interface OnGetPromotionEligibilityListener {
    void onGetPromotionEligibility(@NotNull ErrorVo var1, @NotNull ArrayList<PromotionEligibilityVo> var2);
}
```

And thus, it's allowed by the API to receive multiple `PromotionEligibilityVo`s for the same product ID. I haven't seen this happen in my testing, but since it's technically allowed, it's a case we need to handle.